draft-ietf-quic-version-negotiation.md

```diff
@@ -387,6 +387,40 @@ lifetime of the connection; endpoints MUST NOT change the version that they
 consider to be in use based on the Version field of long headers as that field
 could be forged by attackers.
 
+# Minimum support
+
+Hosts can support this standard incrementally.
+
+{{INV}} and {{QUIC}} already describe how servers send Version Negotiation
+packets in response to long header packets with unsupported version fields, even
+if they support only one version. Doing so prevents long timeouts at the client,
+but servers might not send the packet multiple times for the same connection, or
+when under stress.
+
+If a server supports two or more QUIC versions and sends Version Negotiation
+packets, it MUST send the Version Information Transport Parameter {{vers-info}},
+and parse the client's transport parameter in accordance with {{downgrade}}.
+Failure to comply would leave the connection vulnerable to downgrade attacks, or
+cause connection failure, if the client initiates Incompatible Version
+Negotiation.
+
+A server that supports two or more compatible QUIC versions MAY implement
+Compatible version negotiation if it prefers one version over another.
+
+A client that supports two or more versions, and chooses to respond to a Version
+Negotiation packet with a connection attempt using one of the advertised
+versions, MUST include a Version Information Transport Parameter and process the
+server's transport parameter in accordance with {{downgrade}}.
+
+If a client sends a Version Information Transport Parameter in a packet using a
+version that is compatible with a version advertised in the transport parameter,
+it MUST support compatible version negotiation.
+
+Therefore, clients can only avoid support for compatible negotiation by either
+(1) never attempting connection based on a Version Negotiation packet, and
+electing not to include the transport parameter; or (2) never including a
+supported version in that transport parameter with which the one in use is
+compatible.
```

Comment:
```suggestion
compatible version in the transport parameter.
```

Reply:
Struggled with the commutative property here, thanks.